test/map-file-comment.js

```diff
@@ -4,11 +4,88 @@
+      if (err) {
+        reject(err)
+      } else {
+        resolve(content)
+      }
+    })
+  })
+}
+function readMapAsyncBuffer(filename) {
+  return new Promise(function (resolve, reject) {
+    var filepath = path.join(__dirname, 'fixtures', filename)
+    fs.readFile(filepath, function (err, content) {
+      if (err) {
+        reject(err)
+      } else {
+        resolve(content)
+      }
+    })
+  })
+}
+
+test('\nresolving a "/*# sourceMappingURL=map-file-comment.css.map*/" style comment inside a given css content', function (t) {
+  var css = fs.readFileSync(__dirname + '/fixtures/map-file-comment.css', 'utf8')
+  var conv = convert.fromMapFileSource(css, readMapSyncString);
+  var sm = conv.toObject();
+
+  t.deepEqual(
+      sm.sources
+    , [ './client/sass/core.scss',
+        './client/sass/main.scss' ]
+    , 'resolves paths of original sources'
+  )
+
+  t.equal(sm.file, 'map-file-comment.css', 'includes filename of generated file')
+  t.equal(
+      sm.mappings
+    , 'AAAA,wBAAyB;EACvB,UAAU,EAAE,IAAI;EAChB,MAAM,EAAE,KAAK;EACb,OAAO,EAAE,IAAI;EACb,aAAa,EAAE,iBAAiB;EAChC,KAAK,EAAE,OAAkB;;AAG3B,wBAAyB;EACvB,OAAO,EAAE,IAAI;;ACTf,gBAAiB;EACf,UAAU,EAAE,IAAI;EAChB,KAAK,EAAE,MAAM;;AAGf,kBAAmB;EACjB,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,UAAU,EAAE,KAAK;EACjB,aAAa,EAAE,GAAG;EAClB,KAAK,EAAE,KAAK;;AAEd,kBAAmB;EACjB,KAAK,EAAE,KAAK;;AAGd,mBAAoB;EAClB,KAAK,EAAE,KAAK;EACZ,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,SAAS,EAAE,IAAI'
+    , 'includes mappings'
+  )
+  t.end()
+})
+
+test('\nresolving a "//# sourceMappingURL=map-file-comment.css.map" style comment inside a given css content', function (t) {
+  var css = fs.readFileSync(__dirname + '/fixtures/map-file-comment-double-slash.css', 'utf8')
+  var conv = convert.fromMapFileSource(css, readMapSyncString);
+  var sm = conv.toObject();
+
+  t.deepEqual(
+      sm.sources
+    , [ './client/sass/core.scss',
+        './client/sass/main.scss' ]
+    , 'resolves paths of original sources'
+  )
+
+  t.equal(sm.file, 'map-file-comment.css', 'includes filename of generated file')
+  t.equal(
+      sm.mappings
+    , 'AAAA,wBAAyB;EACvB,UAAU,EAAE,IAAI;EAChB,MAAM,EAAE,KAAK;EACb,OAAO,EAAE,IAAI;EACb,aAAa,EAAE,iBAAiB;EAChC,KAAK,EAAE,OAAkB;;AAG3B,wBAAyB;EACvB,OAAO,EAAE,IAAI;;ACTf,gBAAiB;EACf,UAAU,EAAE,IAAI;EAChB,KAAK,EAAE,MAAM;;AAGf,kBAAmB;EACjB,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,UAAU,EAAE,KAAK;EACjB,aAAa,EAAE,GAAG;EAClB,KAAK,EAAE,KAAK;;AAEd,kBAAmB;EACjB,KAAK,EAAE,KAAK;;AAGd,mBAAoB;EAClB,KAAK,EAAE,KAAK;EACZ,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,SAAS,EAAE,IAAI'
+    , 'includes mappings'
+  )
+  t.end()
+})
+
 test('\nresolving a "/*# sourceMappingURL=map-file-comment.css.map*/" style comment inside a given css content', function (t) {
   var css = fs.readFileSync(__dirname + '/fixtures/map-file-comment.css', 'utf8')
-  var conv = convert.fromMapFileSource(css, __dirname + '/fixtures');
+  var conv = convert.fromMapFileSource(css, readMapSyncBuffer);
```

Comment:
Ideally we'd get the old tests to work without changing them after applying the changes I recommended above.
And on top we can add tests that show that using a _read_ function here works as well.

This applies to all tests below that changed in a similar manner.